tools/test/audio/test_utils/dither_and_quantize.m

```diff
@@ -51,8 +51,7 @@
 nx = sx(1);
 nch = sx(2);
 
-xq = int32(floor( scale*x + randn(nx,nch) - randn(nx,nch) -0.5)); % TPDF dither
-%xq = int32(round( scale*x )); % Omit dither
+xq = int32(round(scale * x + rand(nx,nch) + rand(nx, nch) - 1)); % TPDF dither
```

Comment:
How much of the noise in the triangles is altered here?
Are we increasing the decoupling of the noise floor (dB)?

Reply:
I did this quick compare of noise levels (dBFS) of two dithers with Octave

```
>> nx = 100000;
>> nch = 1;
>> d1 = randn(nx,nch) - randn(nx,nch);
>> d2 = rand(nx,nch) + rand(nx, nch) - 1;
>> mean(d1)
ans = -5.9594e-03
>> mean(d2)
ans = 7.1481e-04
>> 20*log10(sqrt(mean((d1/scale).^2)))
ans = -87.294
>> 20*log10(sqrt(mean((d2/scale).^2)))
ans = -98.091

```

The noise energy in Gaussian is higher and the delta is similar to what I got in measurement.

Reply:
Thank you!!
A tool for assessing RPDF, TPDF Dither, and comparing sound quality is required since dither is a sort of noise that is intentionally introduced to randomize quantization error, making it quite challenging.

Reply:
Small mistake above, both dB values need +3.01 dB add for the used dBFS definition, so the correct dither levels are -84.3 and -95.1 dBFS.

Reply:
Could we implement dither influence in quantization.
Ideally, you should consider each sample after that as a variation from the value before it.

example :
`b tpdf(2:end) = b(2:end)-b(1:end-1); `

Reply:
> Could we implement dither influence in quantization. Ideally, you should consider each sample after that as a variation from the value before it.
> 
> example : `b tpdf(2:end) = b(2:end)-b(1:end-1); `

I don't understand you proposal - above looks like FIR filtering operation if b() are samples?

Reply:
> `rms` should also do mean square root

Results look the same, I used 20*log10(sqrt(mean(x.^2)))  in the example to show the RMS calculation equation.

Reply:
> > Could we implement dither influence in quantization. Ideally, you should consider each sample after that as a variation from the value before it.
> > example : `b tpdf(2:end) = b(2:end)-b(1:end-1); `
> 
> I don't understand you proposal - above looks like FIR filtering operation if b() are samples?

Okay, I believe the plan would benefit from more work on my part, and I believe the current change are acceptable within the scope of the PR.